shell/app-shell/elements/arc-host.js

```diff
@@ -140,7 +128,19 @@ class ArcHost extends Xen.Debug(Xen.Base, log) {
       context: state.context,
       storageKey: state.storageKey
     };
-    const arc = await this._constructArc(state.id, serialization, params);
+    let arc;
+    try {
+      arc = await this._constructArc(state.id, serialization, params);
+    } catch (x) {
+      warn('failed to deserialize arc, will retry');
```

Comment:
Two thoughts:
1. Maybe we should log `x`
2. Maybe it's not a warning-level error? Since I think we expect to see it not infrequently?

Reply:
Because of (2) I kind of didn't want to do (1). I guess I wanted to `warn` for some time because there may be unintended consequences.